ui/app/mainui/Handlers/HandlersManager.qml

```diff
@@ -0,0 +1,141 @@
+        getStatusTokenKey: function() {
+            return root.rootStore.profileSectionStore.ensUsernamesStore.getStatusTokenKey()
+        }
+
+        // for sticker flows
+        stickersMarketAddress: root.rootChatStore.stickersStore.getStickersMarketAddress()
+        stickersNetworkId: root.rootChatStore.appNetworkId
+
+        simpleSendEnabled: root.featureFlagsStore.simpleSendEnabled
+
+        // for simple send
+        walletAccountsModel: root.walletRootStore.accounts
+        filteredFlatNetworksModel: root.networksStore.activeNetworks
+        flatNetworksModel: root.networksStore.allNetworks
+        areTestNetworksEnabled: root.networksStore.areTestNetworksEnabled
+        groupedAccountAssetsModel: root.walletAssetsStore.groupedAccountAssetsModel
+        plainTokensBySymbolModel: root.tokensStore.plainTokensBySymbolModel
+        showCommunityAssetsInSend: root.tokensStore.showCommunityAssetsInSend
+        collectiblesBySymbolModel: root.walletRootStore.collectiblesStore.jointCollectiblesBySymbolModel
+        savedAddressesModel: root.walletRootStore.savedAddresses
+        recentRecipientsModel: root.transactionStore.tempActivityController1Model
+
+        isDetailedCollectibleLoading: root.walletCollectiblesStore.isDetailedCollectibleLoading
+        detailedCollectible: root.walletCollectiblesStore.detailedCollectible
+
+        currentCurrency: root.currencyStore.currentCurrency
+        fnFormatCurrencyAmount: root.currencyStore.formatCurrencyAmount
+        fnFormatCurrencyAmountFromBigInt: root.currencyStore.formatCurrencyAmountFromBigInt
+
+        fnResolveENS: function(ensName, uuid) {
+            root.rootStore.resolveENS(ensName, uuid)
+        }
+
+        fnGetDetailedCollectible: function(chainId, contractAddress, tokenId) {
+            root.walletCollectiblesStore.getDetailedCollectible(chainId, contractAddress, tokenId)
+        }
+
+        fnResetDetailedCollectible: function() {
+            root.walletCollectiblesStore.resetDetailedCollectible()
+        }
+
+        fnGetOpenSeaUrl: function(networkShortName) {
+
+            return root.walletRootStore.getOpenSeaUrl(networkShortName)
+        }
+
+        onLaunchBuyFlowRequested: {
+            buyFormData.selectedWalletAddress = accountAddress
+            buyFormData.selectedNetworkChainId = chainId
+            buyFormData.selectedTokenKey = tokenKey
+            Global.openBuyCryptoModalRequested(buyFormData)
+        }
+
+        Component.onCompleted: {
+            // It's requested from many nested places, so as a workaround we use
+            // Global to shorten the path via global signal.
+            Global.sendToRecipientRequested.connect(sendToRecipient)
+            root.rootStore.ensNameResolved.connect(ensNameResolved)
+        }
+    }
+
+    readonly property StatusGifPopupHandler statusGifPopupHandler: StatusGifPopupHandler {
+        gifStore: sharedRootStore.gifStore
```

Comment:
here as well, can the handler not just ask for not it needs? does the entire gif store ned to be exposed here?

Reply:
Added a new commit for this work